plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/JdbcConnectorExpressionRewriterBuilder.java

```diff
@@ -39,9 +41,14 @@ public static JdbcConnectorExpressionRewriterBuilder newBuilder()
     private JdbcConnectorExpressionRewriterBuilder() {}
 
     public JdbcConnectorExpressionRewriterBuilder addStandardRules(Function<String, String> identifierQuote)
+    {
+        return addStandardRules(identifierQuote, type -> Optional.empty());
+    }
+
+    public JdbcConnectorExpressionRewriterBuilder addStandardRules(Function<String, String> identifierQuote, Function<Type, Optional<String>> typeMapping)
```

Comment:
This is simple for PostgreSQL and varchar case, but generally this is tricky, since remote database and Trino have different type systems.

In case of 
```
where x IN (a, b, c, NULL)
```

the NULL should be eliminated on the engine side (generic engine rule).

Do we have other cases where this is actually useful?

Reply:
`x IN (NULL)` is always false, right? Even for `NULL IN (NULL)` so eliminating NULLs is safe?

also `x IN ()` (empty list) is also false

Reply:
> `x IN (NULL)` is always false, right?

result is `NULL` 


> also `x IN ()` (empty list) is also false

This is not allowed in SQL, but it it was allowed, it would be `false`.

Reply:
> the NULL should be eliminated on the engine side (generic engine rule).

BTW such rule, if written, should go in separate PR.

in this PR please keep the tests exercising `IN` pushdown with NULLs (asserting that these doesn't get pushed down).